### ISSUE TITLE
Request: more modifiers

### ISSUE THREAD

**Issue by kontofikcyjne** [nikitabobko/AeroSpace#587]
Would it be possible to add more modifier keys? For example caps lock, § or fn?

**Comment by kontofikcyjne** [comment]
I think this should be added in this file /AeroSpace/Sources/AppBundle/config/keysMap.swift

**Comment by noahmayr** [comment]
Allowing to individually target left/right shift/alt/cmd would also be helpful imo. There are quite special chars I need alt for and I don't enjoy having to do cramp inducing keyboard acrobatics to do window management.

**Comment by blindfox0** [comment]
> Allowing to individually target left/right shift/alt/cmd would also be helpful imo. There are quite special chars I need alt for and I don't enjoy having to do cramp inducing keyboard acrobatics to do window management.

Exactly! I'm from poland so i have to use alt for letters like ą ć ś etc. Possibility of set for example only left alt for mod key will be great.

**Comment by anschuetz** [comment]
> Allowing to individually target left/right shift/alt/cmd would also be helpful imo. There are quite special chars I need alt for and I don't enjoy having to do cramp inducing keyboard acrobatics to do window management.

This would be #VERY# important: In Germany eg. "@" is alt-L so the window-navigation fails. I could do right-alt-L to @ and left-alt-L to navigate.

**Comment by bcremer** [comment]
A common workaround for non US keyboards is to use a software like https://hyperkey.app/ to map `alt-ctrl-cmd` to the capslock key.

I'm using `alt-ctrl-cmd` and `alt-ctrl-cmd-shift` as my primary modifier-prefix for all shortcuts in aerospace.

**Comment by jo-hb** [comment]
> > Allowing to individually target left/right shift/alt/cmd would also be helpful imo. There are quite special chars I need alt for and I don't enjoy having to do cramp inducing keyboard acrobatics to do window management.
> 
> This would be #VERY# important: In Germany eg. "@" is alt-L so the window-navigation fails. I could do right-alt-L to @ and left-alt-L to navigate.

It isn't only the alt-L key. It's also alt-5 = [ , alt-8 = { etc. that are blocked due to the absence an alternative availability of "fn" which would work perfectly on a German keyboard layout.

**Comment by babytof** [comment]
we have the same exact need for french keyboard where alt is needed for special chars like [ { } ] or | (alt+shift+L)
I already use karabiner to map Caps Lock to Esc and Ctrl, so I cannot use @bcremer proposal :/

**Comment by phischdev** [comment]
Is there a way to alias a combination of modifiers?

Such that I could write something like
`mod1 ~ cmd-option-command`
`mod1 = workspace 1`
`mod1-shift = move-to-workspace 1`
?

**Comment by hxhelm** [comment]
This is definitely a blocker for me, would love to try this out but having no support for differentiating right option from left option does not work with my setup.

I have tried implementing this and got the detection between right & left for the modifiers working, but got stuck because I ended up having to listen to both variants anyway.

The current limitation seems to be that [HotKey](https://github.com/soffes/HotKey) does not support that distinction.

**Comment by szymon3** [comment]
I managed to get it to work for Polish characters by removing workspace bindings that interfere with my special characters (e.g. alt+a, alt+c).
